### DISCUSSION TITLE
I am able to create a miracast receiver, but when trying to cast I get unknown error.

### DISCUSSION THREAD

**Discussion by John-Carroll-AtAsc** [microsoft/WindowsAppSDK#3915] [Q&A]
In WinUI3 I'm able to create a miracast receiver. When I try to cast to it I get an unknown error.
Here is the casting request code:

            CastingDevice castingDevice = await CastingDevice.FromIdAsync(PrimaryCastingDevice.Id);

            _castingConnection = castingDevice.CreateCastingConnection();
            _castingConnection.StateChanged += OnStateChanged;
            _castingConnection.ErrorOccurred += OnErrorOccured;
            _ = await _castingConnection.RequestStartCastingAsync(image.GetAsCastingSource());

Some additional debug log info:

From the Caster:
Found MINIME
Casting device properties:
'WinUI3.BarGui.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.12\System.Text.Encoding.Extensions.dll'. Symbols loaded.
Found [TV] Samsung 8 Series (43)
CastingDevice Id: {57258C12-B416-5737-8CB1-4ACD1E53A0C1}
CastingDevice Friendly Name: MINIME
The thread 0x2420 has exited with code 0 (0x0).
PlaybackType Audio: Supported
PlaybackType Picture: Supported
PlaybackType Video: Supported
PlaybackType None: Supported
ImageToCast source URI: 
CastingConnection connecting
CastingConnection disconnected
CastingConnection connecting
CastingConnection disconnected
CastingConnection error: An unknown error occurred.

From the other machine (receiver):

Miracast session status: Success
MiracastReceiver.Listening status: Listening
MiracastReceiver.WiFi status: MiracastSupported
The thread 0x2da8 has exited with code 0 (0x0).

**Comment by DarranRowe** [comment]
Unfortunately, there are some built in Windows Runtime components that don't work well with desktop applications. The Miracast functionality is still one problematic area.
There are two pieces of documentation to reference to show this. First, is Windows Runtime APIs not supported in desktop apps. There is a section named [Methods that use the Request naming pattern](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-supported-api#methods-that-use-the-request-naming-pattern). If you notice, your code is:
`_ = await _castingConnection.**Request**StartCastingAsync(image.GetAsCastingSource());`
So this falls under this general isn't supported header. This documentation does have problems though, and there are cases like [StoreContext](https://learn.microsoft.com/en-us/uwp/api/windows.services.store.storecontext?view=winrt-22621), interestingly another one documented not to work, which implements a way to use it in desktop applications. Secondly, there is the documentation for CastingConnection and related runtime classes. However, the documentation for CastingConnection, CastingConnection.RequestStartCastingAsync and CastingDevice don't show any indication of this. What's more, CastingConnection.RequestStartCastingAsync states that it must be run on the UI thread, so your use of await wouldn't work anyway.
You could try using IInitializeWithWindow/WinRT.Interop.InitializeWithWindow anyway, but it is likely that this won't work and Miracast functionality just doesn't work with desktop applications right now.

**Reply by John-Carroll-AtAsc** [reply]
Are there any plans to resolve this shortcoming, anytime soon?
